src/SimpleSimulator.cpp

```diff
@@ -688,40 +757,50 @@ int main(int argc, char **argv) {
     }
 
     /* Instantiate inputfiles and set outfile destinations*/
-    std::cerr << "Creating and staging input files plus set destination of output files..." << std::endl;
-    for (auto wms: workload_execution_controllers) {
-        try {
-            for (auto &job_spec: wms->get_workload_spec()) {
-                std::shuffle(job_spec.second.infiles.begin(), job_spec.second.infiles.end(), SimpleSimulator::gen); // Shuffle the input files
-                // Compute the job's incremental inputfiles size
-                double incr_inputfile_size = 0.;
-                for (auto const &f : job_spec.second.infiles) {
-                    incr_inputfile_size += f->getSize();
+    std::cerr << "Creating and staging input files" << std::endl;
+    try {
+        for (auto dss : dataset_specs)
```

Comment:
I see why you changed this from job-specs to datasets. But wouldn't it be safer to keep the loop over jobs, when keeping future changes in mind? That way, it would be always tied to the jobs' input-files. So when only parts of the datasets are processed, the cache-hitrate at simulation start would still be reasonable.